packages/dd-trace/src/appsec/index.js

```diff
@@ -28,6 +28,18 @@ const { storage } = require('../../../datadog-core')
 let isEnabled = false
 let config
 
+function sampleRequest ({ enabled, requestSampling }) {
+  if (!enabled) {
+    return false
+  }
+
+  if (!requestSampling) {
+    return false
+  }
+
+  return Math.random() <= requestSampling
```

Comment:
Do we have to use Math.random ? only analyzing once every x request would be the same outcome in average and way faster than computing a mersenne twister every request no ?

Reply:
V8 employs Xoroshiro128+ (a successor for xorshit algorithm) for generating the random numbers which is way faster than MT algorithm. A quick test executing it a million times rendered 7.012 ms in my machine (7ns per operation). If you still think that it's worth to improve it we can think of another solution.

Reply:
Oh ok cool, I guess my informations were very outdated :+1: